src/AWS.Deploy.ServerMode.Client/Exceptions/PortUnavailableException.cs

```diff
@@ -0,0 +1,17 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.\r
+// SPDX-License-Identifier: Apache-2.0
+
+using System;
+
+namespace AWS.Deploy.ServerMode.Client.Exceptions
+{
+    /// <summary>
+    /// Exception is thrown when deployment tool server failed to start due to unavailability of free ports.
+    /// </summary>
+    public class PortUnavailableException : Exception
+    {
+        public PortUnavailableException(string message): base(message)
```

Comment:
A similar exception is available here https://github.com/aws/aws-dotnet-deploy/blob/main/src/AWS.Deploy.CLI/Exceptions.cs#L69 and created for the same reason. We should reuse instead of replicating these things.

Reply:
`AWS.Deploy.ServerMode.Client.csproj` doesn't (can't) reference CLI. It is an independent package which will be used by the VS Toolkit.